xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -82,6 +83,8 @@ public DBListClass()
             if (item != null) {
                 if (item instanceof String) {
                     result.add(new ListItem((String) item));
+                } else if (item instanceof Number) {
```

Comment:
This IF does the same thing as the one above: creates a new ``ListItem`` with the string value of ``item`` and adds it to the result. They could be merged.